plugins/search/src/plugin.ts

```diff
@@ -25,6 +25,18 @@ import {
   createComponentExtension,
   fetchApiRef,
 } from '@backstage/core-plugin-api';
+import { ReactNode } from 'react';
+import { SearchResultSet } from '@backstage/plugin-search-common';
+import { SidebarSearchModalProps } from './components/SidebarSearchModal';
+
+/**
+ * @public
+ */
+export type SearchResultChildrenProvider = {
+  searchResultChildren?:
+    | ReactNode
+    | ((resultSet: SearchResultSet) => JSX.Element);
+};
```

Comment:
I recommend defining this in the props of the component where it is used, in this case, the `SearchModalProps` type.

As mentioned in the other comments people completely customizing the search modal will define search results manually and the `SidebarSearchModalProps` can pick the `resultItems` type from the `SearchModalProps`.

```suggestion
import { SidebarSearchModalProps } from './components/SidebarSearchModal';

```